src/journal/test-journal-send.c

```diff
@@ -5,11 +5,20 @@
 #include <unistd.h>
 
 #include "sd-journal.h"
-
+#include "fileio.h"
 #include "macro.h"
 #include "memory-util.h"
 
-int main(int argc, char *argv[]) {
+static void test_journal_print(void) {
+        assert_se(sd_journal_print(LOG_INFO, "XXX") == 0);
+        assert_se(sd_journal_print(LOG_INFO, "%s", "YYY") == 0);
+        assert_se(sd_journal_print(LOG_INFO, "X%4094sY", "ZZZ") == 0);
+        assert_se(sd_journal_print(LOG_INFO, "X%*sY", LONG_LINE_MAX - 8 - 4, "ZZZ") == 0);
+        /* Commenting out below test case, it is causing SIGABRT */
```

Comment:
I guess the case needs to be fixed instead of commenting out - obviously something is amiss. The original issue was:

```
--- command ---
18:13:13 SYSTEMD_LANGUAGE_FALLBACK_MAP='/build/src/locale/language-fallback-map' PATH='/systemd-meson-build:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl' SYSTEMD_KBD_MODEL_MAP='/build/src/locale/kbd-model-map' /systemd-meson-build/test-journal-send
--- stderr ---
Assertion 'sd_journal_print(LOG_INFO, "X%*sY", LONG_LINE_MAX - 8 - 3, "ZZZ") == -ENOBUFS' failed at src/journal/test-journal-send.c:17, function test_journal_print(). Aborting.
-------
```

Reply:
When i pass LONG_LINE_MAX size of message from test case then it is getting SIGABRT But Control is not even reaching to sd_journal_printv_with_location() function or any other function which i modified.

I think passing  LONG_LINE_MAX size of message is not handled properly in some other part of the code.

Reply:
It actually does reach `sd_journal_printv_with_location()`

```gdb
(gdb) run
The program being debugged has been started already.
Start it from the beginning? (y or n) y
Starting program: /home/mrc0mmand/repos/systemd/build/test-journal-send 
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".

Breakpoint 2, test_journal_print () at ../src/journal/test-journal-send.c:18
18              assert_se(sd_journal_print(LOG_INFO, "X%*sY", LONG_LINE_MAX - 8 - 3, "ZZZ") == -ENOBUFS);
(gdb) s
sd_journal_print_with_location (priority=6, file=0x402028 "CODE_FILE=../src/journal/test-journal-send.c", line=0x402016 "CODE_LINE=18", func=0x402680 <__func__.3> "test_journal_print", format=0x402010 "X%*sY")
    at ../src/journal/journal-send.c:442
442     _public_ int sd_journal_print_with_location(int priority, const char *file, const char *line, const char *func, const char *format, ...) {
(gdb) 
446             va_start(ap, format);
(gdb) 
447             r = sd_journal_printv_with_location(priority, file, line, func, format, ap);
(gdb) 
sd_journal_printv_with_location (priority=6, file=0x402028 "CODE_FILE=../src/journal/test-journal-send.c", line=0x402016 "CODE_LINE=18", func=0x402680 <__func__.3> "test_journal_print", format=0x402010 "X%*sY", ap=0x7fffffffdb30)
    at ../src/journal/journal-send.c:453
453     _public_ int sd_journal_printv_with_location(int priority, const char *file, const char *line, const char *func, const char *format, va_list ap) {
(gdb) 
455             char sbuf[LINE_MAX + 8] = "MESSAGE=";
(gdb) 
459             char *buffer = sbuf;
(gdb) fin
Run till exit from #0  sd_journal_printv_with_location (priority=6, file=0x402028 "CODE_FILE=../src/journal/test-journal-send.c", line=0x402016 "CODE_LINE=18", func=0x402680 <__func__.3> "test_journal_print", format=0x402010 "X%*sY", 
    ap=0x7fffffffdb30) at ../src/journal/journal-send.c:459
0x00007ffff7ef22ca in sd_journal_print_with_location (priority=6, file=0x402028 "CODE_FILE=../src/journal/test-journal-send.c", line=0x402016 "CODE_LINE=18", func=0x402680 <__func__.3> "test_journal_print", format=0x402010 "X%*sY")
    at ../src/journal/journal-send.c:447
447             r = sd_journal_printv_with_location(priority, file, line, func, format, ap);
Value returned is $2 = 0
(gdb) fin
Run till exit from #0  0x00007ffff7ef22ca in sd_journal_print_with_location (priority=6, file=0x402028 "CODE_FILE=../src/journal/test-journal-send.c", line=0x402016 "CODE_LINE=18", func=0x402680 <__func__.3> "test_journal_print", 
    format=0x402010 "X%*sY") at ../src/journal/journal-send.c:447
0x000000000040126b in test_journal_print () at ../src/journal/test-journal-send.c:18
18              assert_se(sd_journal_print(LOG_INFO, "X%*sY", LONG_LINE_MAX - 8 - 3, "ZZZ") == -ENOBUFS);
Value returned is $3 = 0
(gdb) step
log_assert_failed_realm (realm=LOG_REALM_SYSTEMD, text=0x402088 "sd_journal_print(LOG_INFO, \"X%*sY\", LONG_LINE_MAX - 8 - 3, \"ZZZ\") == -ENOBUFS", file=0x402063 "src/journal/test-journal-send.c", line=18, 
    func=0x4026a0 <__PRETTY_FUNCTION__.2> "test_journal_print") at ../src/basic/log.c:816
816             (void) log_open();
```

But in this very case the overall result of the `sd_journal_print()` call is `0` instead of `-ENOBUFS`.